### PR TITLE
Fixing bitwise comparisons with QC data

### LINKED ISSUE #408: QC bitwise comparison of float values
Some xarray methods may cast the QC variables from type int to type float. For example .resample() will convert all variables to float type. This causes issues when performing bitwise comparisons of QC variables. In particular 

in function get_qc_test_mask() 

    check_bit = set_bit(0, test_number) & qc_variable

Need to check variable type and cast to int before performing bitwise comparison.

### PR DESCRIPTION
**Pull request by kenkehoe**
Closes #408

### CODE REVIEW

**Comment by mgrover1** [comment]
@kenkehoe just let me know when this is ready for review!

**Comment by kenkehoe** [comment]
@mgrover1  I think this is ready for review.

File: act/qc/qcfilter.py
Comment (mgrover1) [inline act/qc/qcfilter.py:280]: This seems like a more robust check than the previous implementation... great!

**Review by mgrover1 — APPROVED** [review]
Looks great to me! Thanks for submitting this fix! The tests passed so this should be ready to merge when you are ready.

**Comment by mgrover1** [comment]
@kenkehoe I edited the text at the top to directly reference #408 so that once this is merged, it'll close that issue.

**Comment by kenkehoe** [comment]
@mgrover1  I believe this is ready for merge.

**Pull request merged**
